assets/js/SignGroups.tsx

```diff
@@ -1,7 +1,361 @@
 import * as React from 'react';
+import ReactDatePicker from 'react-datepicker';
+import { stationConfig, arincToRealtimeId } from './mbta';
+import VirtualSign from './VirtualSign';
+import CustomTextInput from './CustomTextInput';
+import AlertPicker from './AlertPicker';
+import {
+  RouteAlerts,
+  StationConfig,
+  Zone,
+  RouteSignGroups,
+  SignGroup,
+} from './types';
+import { defaultZoneLabel } from './helpers';
 
-function SignGroups(): JSX.Element | null {
-  return <div>Sign Groups Here</div>;
+interface ZoneSelectorProps {
+  config: StationConfig;
+  zone: Zone;
+  line: string;
+  selectedSigns: { [sign: string]: boolean };
+  onSignChange: (signId: string, selected: boolean) => void;
+  readOnly: boolean;
+  kind: 'default' | 'named';
+}
+
+function ZoneSelector({
+  config,
+  zone,
+  line,
+  selectedSigns,
+  onSignChange,
+  readOnly,
+  kind,
+}: ZoneSelectorProps): JSX.Element | null {
+  const handleChange = React.useCallback(
+    (evt) => {
+      if (!readOnly) {
+        onSignChange(evt.target.name, evt.target.checked);
+      }
+    },
+    [onSignChange, readOnly],
+  );
+
+  const zoneConfig = config.zones[zone];
+
+  if (
+    zoneConfig === undefined ||
+    (zoneConfig.label === undefined && kind === 'named') ||
+    (zoneConfig.label !== undefined && kind === 'default')
+  ) {
+    return null;
+  }
+
+  const zoneLabel = zoneConfig.label || defaultZoneLabel(zone);
+  const signId = arincToRealtimeId(`${config.id}-${zone}`, line);
+  const isSelected = selectedSigns[signId] || false;
+
+  return (
+    <div>
+      <label
+        className={`sign_groups--zone_selector${
+          isSelected ? ' sign_groups--zone_selector-selected' : ''
+        }`}
+      >
+        <input
+          type="checkbox"
+          name={signId}
+          checked={isSelected}
+          onChange={handleChange}
+        />
+        {zoneLabel}
+      </label>
+    </div>
+  );
+}
+
+interface SignGroupsFormProps {
+  line: string;
+  currentTime: number;
+  alerts: RouteAlerts;
+  setSignGroup: (line: string, ts: number, signGroup: SignGroup) => void;
+  readOnly: boolean;
+}
+
+function SignGroupsForm({
+  line,
+  currentTime,
+  alerts,
+  setSignGroup,
+  readOnly,
+}: SignGroupsFormProps): JSX.Element | null {
+  const [selectedSigns, setSelectedSigns] = React.useState<{
+    [sign: string]: boolean;
+  }>({});
+  const [line1, setLine1] = React.useState('');
+  const [line2, setLine2] = React.useState('');
+  const [expireDT, setExpireDT] = React.useState<Date | null>(null);
```

Comment:
`expireAt` might be a more conventional name for a datetime value; though I understood `DT` fine, so I don't know if it matters much.